GDevelop.js/scripts/generate-dts.mjs

```diff
@@ -0,0 +1,214 @@
+// @ts-check
+import { readFileSync, writeFileSync } from 'fs';
+import { dirname } from 'path';
+import { fileURLToPath } from 'url';
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+const bindingsFile = readFileSync(
+  __dirname + '/../Bindings/Bindings.idl',
+  'utf-8'
+);
+
+const PrimitiveTypes = new Map([
+  ['DOMString', 'string'],
+  ['long', 'number'],
+  ['double', 'number'],
+  ['void', 'void'],
+  ['boolean', 'boolean'],
+]);
+
+const none = Symbol('No default value');
+class Parser {
+  static i = 0;
+  /** @private */
+  static str = '';
+  /** @private */
+  static l = 0;
```

Comment:
nit but important for future code:
Avoid single letters. They are a pain to grep/replace, you can't multiselect them, and I'm not sure what is `i`, nor `l`, what is `c`, what is `cc` - I have to read the code instead of the name giving the answer to me.

Reply:
Ah yes, I meant to only use these while developing and to change them before committing, but I forgot. Thanks for reminding me!